samples/go/ipfs-chat/main.go

```diff
@@ -491,3 +498,21 @@ func expandRLimit() {
 	d.Chk.NoError(err)
 	fmt.Println("current thread limit:", rLimit)
 }
+
+func reconfigureIPFSChunkStore(sp spec.Spec, topic string, nodeIdx int) (*core.IpfsNode, chunks.ChunkStore) {
+	ds := sp.GetDataset()
```

Comment:
Why bother getting the dataset if you're just going to close it immediately? `sp` already has `DatabaseName` parsed, which is all you need.

Reply:
Done. Much better.